analytics/pubstack/pubstack_module.go

```diff
@@ -0,0 +1,246 @@
+package pubstack
+
+import (
+	"fmt"
+	"github.com/prebid/prebid-server/analytics/pubstack/eventchannel"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/analytics/pubstack/helpers"
+
+	"github.com/prebid/prebid-server/analytics"
+)
+
+type Configuration struct {
+	ScopeId  string          `json:"scopeId"`
+	Endpoint string          `json:"endpoint"`
+	Features map[string]bool `json:"features"`
+}
+
+// routes for events
+const (
+	auction    = "auction"
+	cookieSync = "cookiesync"
+	amp        = "amp"
+	setUID     = "setuid"
+	video      = "video"
+)
+
+type bufferConfig struct {
+	timeout time.Duration
+	count   int64
+	size    int64
+}
+
+type PubstackModule struct {
+	eventChannels map[string]*eventchannel.EventChannel
+	configCh      chan *Configuration
+	scope         string
+	cfg           *Configuration
+	buffsCfg      *bufferConfig
+	mux           sync.Mutex
+}
+
+func NewPubstackModule(scope, endpoint, configRefreshDelay string, maxEventCount int, maxByteSize, maxTime string) (analytics.PBSAnalyticsModule, error) {
+	glog.Infof("[pubstack] Initializing module scope=%s endpoint=%s\n", scope, endpoint)
+
+	// parse args
+
+	refreshDelay, err := time.ParseDuration(configRefreshDelay)
+	if err != nil {
+		return nil, fmt.Errorf("fail to parse the module args, arg=analytics.pubstack.configuration_refresh_delay, :%v", err)
+	}
+
+	bufferCfg, err := newBufferConfig(maxEventCount, maxByteSize, maxTime)
+	if err != nil {
+		return nil, fmt.Errorf("fail to parse the module args, arg=analytics.pubstack.buffers, :%v", err)
+	}
+
+	defaultFeatures := make(map[string]bool)
```

Comment:
Style Option: You may also write this as a literal map initializer, if you prefer:
```
defaultFeatures := map[string]bool{
	auction: false,
	video: false,
	amp: false,
	cookieSync: false,
	serUID: false,
}
```